packages/edit-navigation/src/store/actions.js

```diff
@@ -195,14 +196,29 @@ function computeCustomizedAttribute( blocks, menuId, menuItemsByClientId ) {
 		);
 	}
 
-	function linkBlockToRequestItem( block, parentId, position ) {
+	function blockToRequestItem( block, parentId, position ) {
 		const menuItem = omit( getMenuItemForBlock( block ), 'menus', 'meta' );
+
+		let attributes;
+
+		if ( block.name === 'core/navigation-link' ) {
+			attributes = {
+				type: 'custom',
+				title: block.attributes?.label,
+				original_title: '',
+				url: block.attributes.url,
+			};
+		} else {
+			attributes = {
+				type: 'html',
+				content: serialize( block ),
+			};
+		}
```

Comment:
It's totally too early for that, but a future refactor into a factory-like structure for transform/reverse transform could be a good idea to avoid a huge if/else mess.

Reply:
![I'm so excited because I love mess.](https://media.giphy.com/media/5YiRHZtcSeiEyOpSV7/giphy.gif)